### PR TITLE
add metadata to contract storage

### PR DESCRIPTION
**Pull request by ashneverdawn**
Resolves https://github.com/pendulum-chain/pendulum-ink-wrapper/issues/2

Also, I noticed that the total_supply method was missing (the underlying function was there, but it was not being exposed), so I added that too. And I rearraged the order of the methods a bit to make them in the same order as they are listed in the ERC20 standard, so that they are easier to track and make sure we aren't missing any.

### CODE REVIEW

**Review by TorstenStueber — APPROVED** [review]
Looks good

File: psp_pendulum/lib.rs [hunk 1]
```diff
@@ -181,52 +207,13 @@ mod my_psp22_pallet_asset {
...
-            );
-
-            match result {
-                Result::<(), psp_pendulum_lib::PalletAssetErr>::Ok(_) => {
-                    Result::<(), PSP22Error>::Ok(())
-                }
-                Result::<(), psp_pendulum_lib::PalletAssetErr>::Err(e) => {
-                    Result::<(), PSP22Error>::Err(PSP22Error::from(e))
-                }
-            }
```
Comment (ebma) [inline psp_pendulum/lib.rs:229]: Why are these removed?

Comment (ashneverdawn) [reply psp_pendulum/lib.rs:229]: it's unused code and not needed.
